transport-native-epoll/src/main/java/io/netty/channel/epoll/AbstractEpollChannel.java

```diff
@@ -137,15 +137,23 @@ protected void doDeregister() throws Exception {
     @Override
     protected final void doBeginRead() throws Exception {
         // Channel.read() or ChannelHandlerContext.read() was called
-        AbstractEpollUnsafe unsafe = (AbstractEpollUnsafe) unsafe();
+        final AbstractEpollUnsafe unsafe = (AbstractEpollUnsafe) unsafe();
         unsafe.readPending = true;
 
-        setFlag(readFlag);
-
         // If EPOLL ET mode is enabled and auto read was toggled off on the last read loop then we may not be notified
         // again if we didn't consume all the data. So we force a read operation here if there maybe more data.
         if (unsafe.maybeMoreDataToRead) {
-            unsafe.epollInReady();
+            // Run epollInReady later because this is consistent with declaring interest with the polling mechanism.
+            // We also set maybeMoreDataToRead to false to prevent executing multiple of these runnables.
+            unsafe.maybeMoreDataToRead = false;
+            eventLoop().execute(new OneTimeTask() {
```

Comment:
If it is a `Runnable` it will have to be wrapped in some sort of QueueNode object every time right? I guess if we are recycling QueueNode it is possible we will re-use?

I will do this as it seems like `OneTimeTask` may go away in the near term future anyways.